### DISCUSSION TITLE
Transparent Linear Gradient

### DISCUSSION THREAD

**Discussion by stevetayloruk** [SixLabors/ImageSharp.Web#299] [Q&A] [answered]
Hi,

I'm trying to add a linear gradient the starts solid and fades to transparent over the top of an existing image but can't seem to achieve the transparancy aspect of it.

Before:
![winter](https://user-images.githubusercontent.com/5065543/212283712-5912882d-5f18-4e15-b215-ba64ec846621.jpg)

After:

![winter-grad](https://user-images.githubusercontent.com/5065543/212283737-8c18c596-ece8-4d6e-b470-7811f69c697c.jpg)

I'm using the LinearGradientBrush and the code works with two solid colours, but I can't seem to get the transparancy working.

TIA

**Comment by JimBobSquarePants** [comment, marked as answer]
Without any code it’s hard to say what you are doing wrong. My guess is though that you’re using a pixel format that does not support alpha transparency.

**Comment by stevetayloruk** [comment]
Here is the code I'm using.  The transparency comes out as black

```
public FormattedImage Process(FormattedImage image, ILogger logger, CommandCollection commands, CommandParser parser, CultureInfo culture)
  {
    var extension = commands.GetValueOrDefault(Logo);

    if (!string.IsNullOrEmpty(extension))
    {
      var linearGradientBrush = new LinearGradientBrush(new Point(0, 0), new Point(0, 100), GradientRepetitionMode.DontFill, new ColorStop(0, Color.Black), new ColorStop(1, Color.Transparent));

      image.Image.Mutate(x =>  { 
        x.Fill(linearGradientBrush);
      }   
    );
    }

    return image;
  }
```

Thanks for helping.

**Reply by JimBobSquarePants** [reply]
Ok so you’re using ImageSharp.Web?

What version?

**Reply by stevetayloruk** [reply]
Yes, the latest version. 2.0.2
```
    <PackageReference Include="SixLabors.ImageSharp" Version="2.1.3" />
    <PackageReference Include="SixLabors.ImageSharp.Drawing" Version="1.0.0-beta15" />
    <PackageReference Include="SixLabors.ImageSharp.Web" Version="2.0.2" />
    <PackageReference Include="SixLabors.ImageSharp.Web.Providers.Azure" Version="2.0.2" />
```

**Reply by JimBobSquarePants** [reply]
What's the value of `IImageWebProcessor.RequiresAlphaAwarePixelFormat`?

**Reply by stevetayloruk** [reply]
Sorry, I can't seem to find this property on IImageWebProcessor and can't see it referenced in the documentation either.  The interface only seems to have `RequiresTrueColorPixelFormat`

**Reply by JimBobSquarePants** [reply]
Ah yes, I forgot I updated that before release. 

That property needs to return `true` so that the pipeline knows we require a 32-bit True Color pixel format pixel format (_8 bits per color component plus an 8-bit alpha channel_). You could be smart about it and only return `true` if your linear gradient contains a transparent color.

**Reply by stevetayloruk** [reply]
Thanks!  yes that did the trick.

Appreciate the help.
